clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -167,7 +167,7 @@
  * called <i>test</i> as described above.
  * <p>
  * The broker will automatically detect failed processes in the <i>test</i> group by using a heartbeat mechanism. The
- * consumer will automatically ping the cluster periodically, which lets the cluster know that it is alive. As long as
+ * consumer will ping the cluster whenever {@link #poll(long)} is called, which lets the cluster know that it is alive. As long as
```

Comment:
Right, there's no other way it can fail to do so.  I'll push that wording up now.